servers/annotations-api/src/test/mutation/highlights-batchwrite.integration.ts

```diff
@@ -0,0 +1,368 @@
+      const result = res.body.data?.batchWriteHighlights;
+
+      const expected = {
+        deleted: [],
+        created: expect.toIncludeSameMembers([
+          {
+            id: expect.toBeString(),
+            version: 2,
+            patch: 'Prow scuttle parrel',
+            quote: 'provost Sail ho shrouds spirits boom',
+            _createdAt: expect.toBePositive(),
+            _updatedAt: expect.toBePositive(),
+          },
+        ]),
+      };
+      expect(result).toEqual(expected);
+    });
+    it('should optionally accept a UUID passed from the client and use for the ID', async () => {
+      const id = uuid();
+      const variables: { input: BatchWriteHighlightsInput } = {
+        input: {
+          create: [
+            {
+              id,
+              itemId: '3',
+              version: 2,
+              patch: 'Prow scuttle parrel',
+              quote: 'provost Sail ho shrouds spirits boom',
+            },
+          ],
+        },
+      };
+      const res = await request(app)
+        .post(graphQLUrl)
+        .set(headers)
+        .send({ query: print(BATCH_WRITE_HIGHLIGHTS), variables });
+      const result = res.body.data?.batchWriteHighlights;
+
+      const expected = {
+        deleted: [],
+        created: expect.toIncludeSameMembers([
+          {
+            id,
+            version: 2,
+            patch: 'Prow scuttle parrel',
+            quote: 'provost Sail ho shrouds spirits boom',
+            _createdAt: expect.toBePositive(),
+            _updatedAt: expect.toBePositive(),
+          },
+        ]),
+      };
+      expect(result).toEqual(expected);
+    });
+    it('should not overrwrite an existing highlight', async () => {
+      const seed = await request(app)
+        .post(graphQLUrl)
+        .set(headers)
+        .send({
+          query: print(CREATE_HIGHLIGHTS),
+          variables: {
+            input: [
+              {
+                id: uuid(),
+                itemId: '1',
+                version: 2,
+                patch: 'Prow scuttle parrel',
+                quote: 'provost Sail ho shrouds spirits boom',
+              },
+            ],
+          },
+        });
+      const seedResult = seed.body.data?.createSavedItemHighlights;
+      expect(seedResult.length).toEqual(1);
+      const id = seedResult[0].id;
+      const variables = {
+        input: {
+          create: [
+            {
+              id,
+              itemId: '1',
+              version: 2,
+              patch: 'Prow scuttle parrel',
+              quote: 'Bring a spring upon her cable holystone',
+            },
+          ],
+        },
+      };
+      const res = await request(app)
+        .post(graphQLUrl)
+        .set(headers)
+        .send({ query: print(BATCH_WRITE_HIGHLIGHTS), variables });
+      const result = res.body.errors;
+      expect(result.length).toBe(1);
+      expect(result[0].extensions.code).toBe('INTERNAL_SERVER_ERROR');
```

Comment:
bleh reminder we need to fix errors :(

Reply:
oh this one is correct -- it's a database error due to duplicate key. well, i guess if we wanted to it could be different.